apps/src/templates/census2017/CensusTeacherBanner.jsx

```diff
@@ -0,0 +1,350 @@
+    backgroundColor: color.white,
+    overflowWrap: 'break-word'
+  },
+  message: {
+    marginTop: 0,
+    marginBottom: 20,
+    marginLeft: 20,
+    marginRight: 20,
+  },
+  radio: {
+    verticalAlign: 'top',
+    marginRight: 10,
+  },
+  title: {
+    marginBottom: 0,
+  },
+  updateSchool: {
+    fontSize: '85%',
+    marginTop: 0,
+    marginBottom: 0,
+  },
+};
+
+export default class CensusTeacherBanner extends Component {
+  static propTypes = {
+    schoolYear: PropTypes.number.isRequired,
+    onSubmit: PropTypes.func.isRequired,
+    onDismiss: PropTypes.func.isRequired,
+    onPostpone: PropTypes.func.isRequired,
+    onChange: PropTypes.func.isRequired,
+    ncesSchoolId: PropTypes.string.isRequired,
+    question: PropTypes.oneOf(['how_many_10_hours', 'how_many_20_hours']).isRequired,
+    selection: PropTypes.bool,
+    teacherId: PropTypes.number.isRequired,
+    teacherName: PropTypes.string.isRequired,
+    teacherEmail: PropTypes.string.isRequired,
+    showUnknownError: PropTypes.bool,
+  };
+
+  constructor() {
+    super();
+    this.bindSchoolInfoInputs = this.bindSchoolInfoInputs.bind(this);
+  }
+
+  componentDidMount() {
+    this.loadSchoolName(this.props.ncesSchoolId);
+  }
+
+  state = {
+    showSchoolInfoForm: false,
+    country: 'United States',
+    schoolType: '',
+    ncesSchoolId: null,
+    schoolName: '',
+    schoolDisplayName: null,
+    schoolState: '',
+    schoolZip: '',
+    schoolLocation: '',
+  };
+
+  onCountryChange(_, event) {
+    const newCountry = event ? event.value : '';
+    this.setState({country: newCountry});
+  }
+
+  onSchoolTypeChange(event) {
+    const newType = event ? event.target.value : '';
+    this.setState({schoolType: newType});
+  }
+
+  onSchoolChange(_, event) {
+    const newSchool = event ? event.value : '';
+    this.setState({ncesSchoolId: newSchool});
+    this.loadSchoolName(newSchool);
+  }
+
+  onSchoolNotFoundChange(field, event) {
+    const newValue = event ? event.target.value : '';
+    this.setState({
+      [field]: newValue
+    });
+  }
+
+  showSchoolInfoForm() {
+    this.setState({showSchoolInfoForm: true});
+  }
+
+  hideSchoolInfoForm() {
+    this.setState({
+      showSchoolInfoForm: false,
+      showSchoolInfoErrors: false,
+    });
+  }
+
+  dismissSchoolInfoForm() {
+    // Clear any state that may have been set
+    this.setState({
+      showSchoolInfoForm: false,
+      country: 'United States',
+      schoolType: '',
+      ncesSchoolId: null,
+      schoolName: '',
+      schoolDisplayName: null,
+      schoolState: '',
+      schoolZip: '',
+      schoolLocation: '',
+      showSchoolInfoErrors: false,
+      showSchoolInfoUnknownError: false,
+    });
+    this.loadSchoolName(this.props.ncesSchoolId);
+  }
+
+  onSchoolInfoSubmit() {
+    if (this.schoolInfoInputs.isValid()) {
+      $.ajax({
+        url: "/users.json",
+        type: "post",
+        dataType: "json",
+        data: $("#census-school-info-form").serialize()
```

Comment:
Since this is all in React now, don't we have (or can construct) all the state in one place in the parent object and submit that? Why construct all these hidden fields (starting on line 276) and serializing the actual html form data? Is there something else we're gaining by doing that? This would be simpler, and more React-y, to maintain and submit state from a component, not render anything hidden (unless it's needed elsewhere), and not rely on finding an element by id. If we were to look for an element by id, using `ref` is generally preferred.

Reply:
I'm going to change this to construct the POST from the component's state. I did it this way simply because all of the other forms I've touched have worked like this and I didn't consider doing it differently. I agree that we don't need an actual form here.